Guide to creating a New Architecture app from template (#3056)
facebook/react-native-website: docs/the-new-architecture/use-app-template.md

---
id: use-app-template
title: Creating a New Architecture App
---

import Tabs from '@theme/Tabs'; import TabItem from '@theme/TabItem'; import constants from '@site/core/TabsConstants';

import RemoveGlobalCLI from '.././\_remove-global-cli.md';

This page will help you create a new React Native app that uses the New Architecture.

## Development environment

Before continuing, make sure you've followed all the steps in [Setting up the development environment](getting-started.md), under the **React Native CLI Quickstart** tab.

If following the setup guide, stop when you reach the section **Running your React Native Application**, and resume following this guide.

> :::note

Reviewer: ```suggestion
:::caution
```

Author: For my future reference, is there a particular reason you want to go with `caution` here? I went with `note` for any that just add extra info, since it felt like a lot of yellow blocks on the page otherwise, very visually busy.

Reviewer: IIRC, note/info box are blue and white. They can slip and the user may not notice them. I think that they are good for additional information that, even if the users don't read them, the output doesn't change very much.

A caution box is yellow, and harder to miss. I think we should use it every time we are delivering information that, if missed, may generate some trouble. For example, if the users configure the New Architecture on version 0.67, they will incur in a lot of errors. And it could be tricky to spot that they are using the wrong version. Hence, the yellow box.
Similarly, if the skipped the `Settings Up Your Environment` guide because they are used to work with Expo and they try to use the New Architecture, it won't work, and a yellow box could catch their attention more easily.

Does this make sense?

Author: I think I have a different opinion about how this would be read -- for me, too many yellow boxes can visually interfere with the ability to skim the page's structure -- but it's a matter of opinion, so happy to stick with this.

> The New Architecture cannot currently be enabled in an Expo app.

Reviewer 2: ```suggestion
If you're using Expo, you can't enable the New Architecture at the moment and have to wait for a future release of the Expo SDK
```

small nit to clarify

:::

## Creating a new application

<RemoveGlobalCLI />

If you already have your development environment set up, create a new React Native project from the template:

```shell
npx react-native init AwesomeProject
```

:::caution

> The New Architecture is available in version 0.68 or later.

Reviewer 2: ```suggestion
The New Architecture is available in React Native version 0.68 or later.
```

---
id: use-app-template
title: Creating a New Architecture App
---

import Tabs from '@theme/Tabs'; import TabItem from '@theme/TabItem'; import constants from '@site/core/TabsConstants';

import RemoveGlobalCLI from '.././\_remove-global-cli.md';

This page will help you create a new React Native app that uses the New Architecture.

## Development environment

> Before continuing, make sure you've followed all the steps in [Setting up the
> development environment](getting-started.md), under the **React Native CLI
> Quickstart** tab.

Reviewer: ```suggestion
Before continuing, make sure you've followed all the steps in [Setting up the development environment](getting-started.md), under the **React Native CLI Quickstart** tab.
```

> If following the setup guide, stop when you reach the section **Running your React Native
> Application**, and resume following this guide.

Reviewer: ```suggestion
If following the setup guide, stop when you reach the section **Running your React Native Application**, and resume following this guide.
```

Author: > A question: are you using some tools to write the guide? I found several sentences with a new line in the middle of the sentence which could lead to bugs (missing whitespaces mainly). It really looks like a sort of word wrap feature. 

Sort of, I manually wrapped paragraphs that were getting very long -- it was just a habit from writing Markdown docs for other purposes, I'll fix it 😬 thanks for catching!

**Note**: The New Architecture cannot currently be enabled in an Expo app.

## Creating a new application

<RemoveGlobalCLI />

If you already have your development environment set up, create a new React Native project from the template:

```shell
npx react-native init AwesomeProject
```

> If specifying a versioned template, use version 0.68 or later.

Reviewer: ```suggestion
**Note:** the New Architecture is available from version 0.68 or later.
```

I found personally confusing the _If specifying a version_. What does that mean? Should we push for using a versioned template? Or in case a user wants to use a versioned template, we are suggesting using version 0.68 or later?

## Configuration

Follow the steps below to enable the New Architecture and build the app.

### Enable Hermes

Hermes is an open-source JavaScript engine optimized for React Native. [Hermes will be the default engine in the future](https://github.com/reactwg/react-native-new-architecture/discussions/4), and we highly recommend you use it.

Please [follow the instructions on the React Native website](hermes.md) in order to enable Hermes in your application.

### Enable the New Architecture

#### Target OS

<Tabs groupId="platform" defaultValue={constants.defaultPlatform} values={constants.platforms} className="pill-tabs">
<TabItem value="ios">

Navigate to the `ios` directory and run the following:

```shell
# from `ios` directory

> RCT_NEW_ARCH_ENABLED=1 bundle exec pod install

Author: @cipolleschi I noticed elsewhere in the docs there's this advice: 

```
Mac M1 architecture is not directly compatible with Cocoapods. If you encounter issues when installing pods, you can solve it by running:
- `sudo arch -x86_64 gem install ffi`
- `arch -x86_64 pod install`
```

But I didn't run into any issues when following instructions without the `arch -x86_64` flag, building this on my M1, as far as I could tell. Do you know if there's a more specific circumstance where this is needed?

Reviewer: The issue why we put that warning is due to an older version of the `ffi` gem. I think that it has been updated in the `0.68`, so we will probably remove it from the doc completely.
Multiple people reported that those instructions are not really needed anymore.

> RCT_NEW_ARCH_ENABLED=1 bundle exec pod install

Author: I opted to go with `bundle exec` here because I've seen several places that say that's what to do going forward, but I noticed 1. No setup instructions exist that tell you to run `bundle install`, but I did need to do that, and 2. I don't think other places in the docs mention using `bundle exec` instead now.

It doesn't seem like a _huge_ deal, if you don't `bundle install` first the `bundle exec` failure will tell you to, but wondering if that's maybe something to follow up on later?

```

Then build and run the app as usual:

```shell
npx react-native run-ios
```

> > **Tip**: You need to re-run `pod install` each time a dependency with native

Reviewer: @cortinico: do we have any `:::`-box to wrap these tips nicely, instead of using quotes?

Reviewer 2: Yes we do. I saw you folks already sorted this out, but in case you're wondering how this feature works for the future https://docusaurus.io/docs/markdown-features/admonitions

> code changes. Make this command easier to run by adding it to `scripts` to your
> project's `package.json` file:
>
> ```
> "scripts": {
>   "pod-install": "RCT_NEW_ARCH_ENABLED=1 bundle exec pod install"
> }
> ```
>

> > and run it with `yarn pod-install`.

Reviewer: ```suggestion
> and run it with `yarn pod-install`. Following `pod install` execution won't require to run `bundle install` again: this instruction should be executed only once.
```

To follow-up from my previous comment.

#### Troubleshooting

##### `react-native run-ios` fails

> If you see a build failure from `react-native run-ios`, there may be cached files from a previous build with
> the old architecture. Clean the build cache and try again:

Reviewer: ```suggestion
If you see a build failure from `react-native run-ios`, there may be cached files from a previous build with the old architecture. Clean the build cache and try again:
```

1. Open the project `ios/project.xcworkspace` in Xcode
2. In XCode, choose Product > Clean Build Folder
3. Try building and running the app again

> 

Reviewer: Does it make sense to explain how to clean up the pods as well? Something like:

```suggestion
4. If the error persists, go to the `ios` folder and run: `rm -rf Pods Podfile.lock build`. This cleans up the pods and the build folder who may be stale. Notice that these instruction requires another execution of `yarn pod-install`. 
```

Author: Yup, good call.

Author: I opted for always recommending removing the Pods; as general troubleshooting advice, this seems reasonable to me?

Author: Also, because of the inclusion of using `yarn pod-install` again, I opted to swap the `npx react-native` calls for their corresponding `yarn` scripts in the template, it seems good to match that where possible, and the `yarn` invocations are much more concise.

Author: > I'm wondering whether we should also explain that the build will take longer after deleting the pods, but I guess it is implied, given that we are purging the cache

Yeah. I also think that for any user who would not expect this, they probably also don't have expectations about build time (i.e., they are new to the ecosystem and trust the tooling). And over time I think this is a pattern that becomes apparent to RN developers, even if they don't know the iOS ecosystem.

</TabItem>
<TabItem value="android">

Set the `newArchEnabled` property to `true` by **either**:

- Changing the corresponding line in `android/gradle.properties`
- Setting the environment variable `ORG_GRADLE_PROJECT_newArchEnabled=true`

Then build and run the app as usual:

```shell
npx react-native run-android
```

> > **Note**: You may notice longer build times with the New Architecture, due to
> > additional step of C++ compilation with the Android NDK. To improve your build
> > time, see
> > [Speeding Up Your Build Phase](docs/build-speed.md).

Reviewer: ```suggestion
:::info
You may notice longer build times with the New Architecture, due to additional step of C++ compilation with the Android NDK. To improve your build time, see [Speeding Up Your Build Phase](docs/build-speed.md).
:::
```

The `:::info` syntax wraps the block in a nicely styled box. There are different alternatives available, but I can't find a reference for them.

Author: Oh good to know, thanks!

</TabItem>
</Tabs>

### Confirming the New Architecture is in use

> After you build and run the app, when Metro serves the JavaScript bundle, you
> should see `"fabric": true` in the Metro logs:

Reviewer: ```suggestion
After you build and run the app, when Metro serves the JavaScript bundle, you should see `"fabric": true` in the Metro logs:
```

<img src="/docs/assets/metro-new-arch.png" alt="Metro shows fabric: true" width="600" />

> ### Want to know more?

Reviewer: I think that the answer is yes. Of course, they have to do the opposite steps that we are suggesting here:
* for iOS: run `RCT_NEW_ARCH_ENABLED=0 pod install` (or just `pod install`, given that the default value for the flag is `0`).
* for Android: switch the `newArchEnabled` and the `ORG_GRADLE_PROJECT_newArchEnabled` to `false` and relaunch the app.

From the Template perspective, this should be enough. The template works with both the old and the New Architecture out-of-the-box. If the devs started adding dependencies that only supports the New Architecture, their app won't work, clearly.

> ### Want to know more?

Author: If it is just switching those flags, I think we can probably leave it alone for now unless anyone's confused by it, but let me know if you think it would be good to add!

> ### Want to know more?

Reviewer: Let's omit that, though. I'd rather have a leaner documentation and it looks reasonable that the users try to turn off those switches to go back to the Old Architecture, even without an explicit paragraph!

If you'd like to view the code changes relevant for the New Architecture, take a

> look at the [upgrade
> helper from version 0.67.4 to 0.68.0](https://react-native-community.github.io/upgrade-helper/?from=0.67.4&to=0.68.0).
> Files that were added for the New Architecture are marked with a yellow banner.

Reviewer: ```suggestion
look at the [upgrade helper from version 0.67.4 to 0.68.0](https://react-native-community.github.io/upgrade-helper/?from=0.67.4&to=0.68.0). Files that were added for the New Architecture are marked with a yellow banner.
```

For further explanations of what each file is doing, check out these guides to walk through the changes step-by-step:

#### Android

> - [Enabling TurboModules on Android](new-architecture-app-modules-android.md)

Author: I noticed the article in the sidebar is titled `Enabling TurboModule on Android` -- I've mostly seen `TurboModules`, plural, though. Is the plural correct?

Reviewer: I think that in general, we refer to `TurboModules`. If we are referring to a specific one, we should use the singular `TurboModule`.

> title: Using the New App Template

Author: I was wondering if there might be a better way to title this. "New App Template" is not necessarily meaningful to an RN developer (externally, this just looks like I'm using the template for 0.68 and changing a configuration flag.)

I was thinking it could be "Creating a New Architecture app", but I'm wondering if there's more information we're trying to convey by describing this as using "the new app template".

This may also relate to my question below about the relation of this page to "Getting Started".

Author: Yup, that all makes sense, thank you!

Reviewer 2: > I was thinking it could be "Creating a New Architecture app", but I'm wondering if there's more information we're trying to convey by describing this as using "the new app template".

+1 here. We always referred to "Using the New App Template" internally to quickly refer to this concept. However "Creating a New Architecture app" is more widely understandable and makes less assumptions.

Reviewer: Feel free to change the title of the page @lindboe

Author: Updated!

---

> This page will help you create a new React Native app that uses the new architecture.

Author: The future tense I copied from "Getting started", trying to keep a consistent tone. Totally happy to change it.

Thank you for the capitalization comment!

> This page will help you create a new React Native app that uses the new architecture.

Reviewer 2: > Thank you for the capitalization comment!

Nit on this: We always (try to) refer to `New Architecture` and `New Architecture Rollout` with titlecase.

As for the tense, I don't have a strong opinion about it 👍

> This page will help you create a new React Native app that uses the new architecture.

Author: Did you have any further thoughts on tense @cipolleschi? I'm still leaning towards keeping it for now since other pages are using it.

> This page will help you create a new React Native app that uses the new architecture.

Reviewer: Good for me. (Sorry I haven't seen this comment yesterday)

## Installing dependencies

> Start by setting up your development environment as described in [Getting Started](getting-started.md), under "React Native CLI Quickstart". Stop when you reach "React Native Command Line Interface", and resume here.

Author: Yeah, unfortunately docusaurus does not support anchor links here. This is originally why I started thinking of ways to combine "Getting Started" with this page.

> Start by setting up your development environment as described in [Getting Started](getting-started.md), under "React Native CLI Quickstart". Stop when you reach "React Native Command Line Interface", and resume here.

Reviewer 2: cc @Simek I've noticed this behavior as well in the website (anchor links on tabbed pages are sometimes broken). Do you know if someone looked into this in the past?

> Start by setting up your development environment as described in [Getting Started](getting-started.md), under "React Native CLI Quickstart". Stop when you reach "React Native Command Line Interface", and resume here.

Author: There's a potential way around this that I found, but I'm not sure if we want it.

If we uniquely title the headers per tab, anchor links to headers will work. The problem is that multiple combinations of tabs have "Installing dependencies" as a header.

We could instead repeat the tab values in the headers, manually, which creates a unique anchor link:

<img width="1461" alt="Screen Shot 2022-04-07 at 3 59 55 PM" src="https://user-images.githubusercontent.com/5252268/162333018-94cca1d4-a054-4449-9de2-0a622e7bf8e2.png">

Obviously that's more verbose. But I think it could work, because:

* The page is somewhat long and when I'm doing a multi-platform setup in a guide like this where I might be doing multiple things at once, it's honestly helpful to be able to have a reminder of which options I'm looking at at the moment without having to scroll up
* We could only do it on main headers, or as needed for linking

But still, less ideal than having a docusaurus fix for this.

> Start by setting up your development environment as described in [Getting Started](getting-started.md), under "React Native CLI Quickstart". Stop when you reach "React Native Command Line Interface", and resume here.

Reviewer 3: It's a known issue, that hash links do not work correctly in Tabs or partials (for tabs they only work if you have correct tabs stored in local storage). Not sure if we can do anything, other that stop using tabs and refactor the content. But we need to keep in mind that part of docs is quite crucial and we need to carefully think about changes to this page.

Refs:
* https://github.com/facebook/docusaurus/issues/7008
* https://docusaurus.canny.io/feature-requests/p/link-to-tab

<h3>React Native Command Line Interface</h3>

> React Native has a built-in command line interface. Rather than install and manage a specific version of the CLI globally, we recommend you access the current version at runtime using `npx`, which ships with Node.js. With `npx react-native <command>`, the current stable version of the CLI will be downloaded and executed at the time the command is run.

Reviewer 2: One point to potentially add here: mention that as of now, you can't use the New Architecture with Expo.

<h2>Creating a new application</h2>

> > If you previously installed a global `react-native-cli` package, please remove it as it may cause unexpected issues.

Author: It would probably be nice for this to include an `npm` example command, but left it as-is for now as it was copied from "Getting Started".

Reviewer 2: Something like:
```
npm uninstall -g react-native
npm uninstall -g @react-native-community/cli
```

(I haven't tested them).

If we add them, let's backport them to the Getting started page 👍

Author: Added

> React Native has a built-in command line interface, which you can use to generate a new project. You can access it without installing anything globally using `npx`, which ships with Node.js. Let's create a new React Native project called "AwesomeProject":

Author: These lines are pulled directly from the "Getting started" section because I noticed there's a lot of overlap in the steps, still, even beyond dependency installation. Comparing this documentation with "Getting started" leads me to a few questions:

* Is it appropriate to repeat all of this information? "Getting started" is very thorough, since it's aimed at beginners, but I would suspect many beginners to React Native will be interested in starting with the new architecture enabled.
* If we're going to be repeating information, should we make reusable markdown fragments to use in both locations? There are a few items that would need to be different (e.g., "if integrating react-native into an existing app" link), but that seems manageable.
* "Getting Started" also includes instructions with command line examples for Windows and Linux. Do we need that support here too? (e.g., some troubleshooting examples might need to be different per-platform)
* Considering all of the above -- should this documentation perhaps also live within "Getting Started"?

Author: > For the time being, we can definitely add a link in the Getting Started guide to this section saying that, if you are interested in creating a new app with the New Architecture, this is the article for you.

Sounds good!

> I don't think we should cover the case of how to integrate RN and the New Architecture in an existing app in this page. 

To make sure it's clear, I just quoted that example because it was part of the page I copied from originally, so yeah, it's an example of something that would have to change if we used snippets.

> If they are related to how to run the app on those platforms, I think yes. Especially if there are difference on how the cli runs on those platforms.

Okay, that may come up for Android, depending on the troubleshooting section. If it does I'll plan to move to using Mac/Windows/Linux platform tabs like "Getting Started".

> But I'm starting thinking that those section should be splitted

To make sure I understand, are you taking about the structure of "Getting Started" here? Or this page?

> Also, the first two sentences are a repetition of the <h3>React Native Command Line Interface</h3> section and we could remove them.

My hope with creating reusable fragments was that we could actually avoid linking to "Getting Started" altogether, and instead copy over the contents of the page that are relevant. I think that might be a better experience, because:
* [there's no way to link to the specific relevant tab in "Getting Started"](https://docusaurus.canny.io/feature-requests/p/link-to-tab), so as-is, the reader is shown the tab for the default Expo CLI Quickstart.
* The directions to "stop reading at when you reach this section and come back" adds a little more friction. The reader will likely forget where to stop reading and may make a mistake and go too far, and get confused about what to do next. Not a huge issue, but perhaps less than ideal.

Reviewer: > To make sure I understand, are you taking about the structure of "Getting Started" here? Or this page?

I'm talking in the `Getting Started` section. It could be split into two:
1. setup your environment
1. create your first app

For the New Architecture, we only have to do the second point, because the first one will be the same, independently from the architecture. This will also relieve you from the burden of ask the user to setup their environment.

> My hope with creating reusable fragments was that we could actually avoid linking to "Getting Started" altogether, and instead copy over the contents of the page that are relevant.

This is a good point. Another [user](https://github.com/reactwg/react-native-new-architecture/discussions/9#discussioncomment-2465185) pointed out a similar issue, so it can be a particularly good solution. But I think it works well for small paragraph: reintroducing the whole setup page could create a big article in the guide that can be intimidating.

For bigger articles, like `Setup your environment` could make sense to add it twice in the sidebar: we are still reusing the material completely and we create a logical flow to follow. What do you think?

Author: > For bigger articles, like Setup your environment could make sense to add it twice in the sidebar: we are still reusing the material completely and we create a logical flow to follow. 

If I'm understanding this correctly, this would mean a left sidebar that looks like:

```
Basics
Environment Setup
  Setting up the development environment
  Creating a new app
  Integration with Existing Apps
  ...
The New Architecture
  Introduction
  Why a New Architecture
  Setting up the development environment
  Creating a new app
```

One obstacle for that: I think the reason "setting up the development environment" and "creating a new app" concepts are combined in the existing "Getting Started" guide is probably because of the Expo guide — it does both at once. I'm not sure how to represent that without Expo being its own page as well, which feels less polished.

Reviewer 2: > For the time being, we can definitely add a link in the _Getting Started_ guide to this section saying that, if you are interested in creating a new app with the New Architecture, this is the article for you.

I don't have a strong opinion here, but I believe we can avoid re-explaining how the whole CLI works and just mention the crucial steps (i.e. the `npx react-native init` and so on).

I think it's safe to assume the user has an environment setup & mention that page as a prerequisite.

Author: Opted for a simpler breakdown here, just linking to the existing guide and mentioning that it's a prerequisite.

If it still looks awkward, I do think the best bet is copying relevant sections of the quickstart guide to here.

```shell

> npx react-native init AwesomeProject

Author: Question: now that 0.68 is out, does this command need to use a particular template? Is it sufficient to just add a note saying "Use 0.68 or later"?

Reviewer: By default, this creates an app using the template shipped with the `0.68`. 
I will specify that we can install different versions of RN using the `--version` switch and that the New Architecture is supported starting from version `0.68`.

What do you think?

Author: That sounds good to me!

```

> ## Configuring iOS

Author: For each of the platform-specific sections, I know we want to include information about the code the new template is generating. Is there any existing documentation/explanation of that? I don't know too much about it myself.

Reviewer: From the configuration perspective, there is nothing to explain here. 

I would add a section specifying what the template creates after the configuration section. For what the template creates, it should basically create a copy of what is into this [folder](https://github.com/facebook/react-native/tree/main/template). Consider that the files that starts with an `_` will be transformed in hidden files (which starts with a `.` instead).

@cortinico Am I missing something important to mention about the template, here?

Author: Yeah, breaking this up into tabs works for me!

Reviewer: Yeah, exactly. I think that all these `#ifdef` and all that code related to the [jsExecutorFactoryForBridge](http://localhost:3000/docs/next/new-architecture-app-modules-ios#2-install-turbomodulemanager-javascript-bindings) or to the [MyComponentRegistry](http://localhost:3000/docs/next/new-architecture-app-modules-ios#2-install-turbomodulemanager-javascript-bindings), for example, could be intimidating for a new react native user.

Author: Agreed. So would we rather skip explaining the native code in this document, then?

Reviewer 2: > @cortinico Am I missing something important to mention about the template, here?

So I would maybe mention (in a box?) the upgrade helper:
https://react-native-community.github.io/upgrade-helper/?from=0.67.4&to=0.68.0
(This is essentially a diff of the new app template of 0.67 and 0.68).

Specifically we added banners there for files that are New Architecture only.

If you check through the diff, you'll see that there are several `.cpp`, and `.h` files inside the `/android` folder. Maybe it's worth mentioning that this new template already sets up all the C++ machinery needed to use the full power of the New Architecture.

If users want to know more about what each file is doing on Android, they're explained here:
https://reactnative.dev/docs/new-architecture-app-modules-android
https://reactnative.dev/docs/new-architecture-app-renderer-android

> we rather skip explaining the native code in this document

I'd say yes. We should mention that there is some native code, maybe mention why it's needed + link to those pages if users are interested on how to "recreate" this template step by step.

> I would add a section specifying what the template creates after the configuration section. For what the template creates, it should basically create a copy of what is into this [folder](https://github.com/facebook/react-native/tree/main/template?rgh-link-date=2022-04-06T10%3A12%3A46Z). Consider that the files that starts with an _ will be transformed in hidden files (which starts with a . instead).

Exactly. @cipolleschi nit: There are a couple of transformation happening as well (package name, app names, app ids, and so on). For the sake of this page, they can be ignored as they're minor implementation details.

Author: I added the relevant links @cortinico mentioned, as well as the corresponding options for iOS.

One thing I don't know yet that I think might be nice to explain is a brief statement on what is happening on each step. E.g., when I set `RCT_NEW_ARCH_ENABLED=1` on `pod install`, briefly, what does that change (e.g., "switches on Code Generation", or for Android, "`react-native run-android` now also compiles C++ code with Android NDK (what C++ code, I'm not sure how to explain?)). But not sure if that was intended only for other more detailed guides?

Reviewer: I think that these are details slightly too deeply nested in how the New Architecture is implemented.
 
I'm not sure that it make sense to explain that the `RCT_NEW_ARCH_ENABLED` for iOS installs additional dependencies required by the New Architecture, generates some C++ code, and set some compilation flags so that the right code is compiled and executed. On Android it does more or less the same, but the compilation flags (which does not exists on that platform). 

But if you see the value there, let's add these short sentences. :+1:

Author: On the one hand, I think it can feel empowering for RN devs to have a high-level summary of what's being changed without having to go read the code.

On the other hand, you're right, it's a little down in the weeds, and if it were likely to change it's probably particularly not useful to know. And given that there's a manual migration guide, that should at least give anyone curious a better idea of the necessary changes.

Reviewer 2: > And given that there's a manual migration guide, that should at least give anyone curious a better idea of the necessary changes.

☝️ You hit the point here. I think we don't need to deep dive too much on what is this extra flag doing, but just mention what it allows you to do. Users can still go to the migration guide to fully understand what is happenign here.

To configure iOS to use the new architecture, navigate to the `ios` directory and run the following:

```shell
# from `ios` directory

> RCT_NEW_ARCH_ENABLED=1 bundle exec pod install

Author: Are there plans to recommend some sort of alias (or maybe include it in the template?) to make this easier to remember?

Reviewer: No, but we could add a box with `pro-tip` or some similar suggestions with that. I think that users love tips & tricks when they saw them.

Author: Added!

```

Then build and run the app as usual:

```shell
npx react-native run-ios
```

### Troubleshooting

If you see a build failure, there may be cached files from a previous build with the old architecture. Clean the build cache and try again:

```shell
# from `ios` directory

> xcodebuild clean

Author: Based off of @cipolleschi 's comment here: https://github.com/reactwg/react-native-new-architecture/discussions/5#discussioncomment-2509370. From what I've seen this command also deletes the Pods folder and Podfile.lock.

I still need to make sure this solves the problem I was having on my old laptop, but it seems good to include regardless, it's just a useful troubleshooting step most of the time.

Reviewer: Amazing idea to add this subsection

RCT_NEW_ARCH_ENABLED=1 bundle exec pod install
```

> ## Configuring Android

Author: I did run into a problem here, but don't know the solution yet, so troubleshooting section to come.

The problem is that on attempting to `run-android` with the new architecture enabled on an M1 laptop, I get:

```* What went wrong:
Execution failed for task ':ReactAndroid:configureNdkBuildDebug'.
...
  ERROR: Unknown host CPU architecture: arm64
```

which looks like an M1-related issue, despite my attempts to use Zulu as the JDK (which appears to have actually been used).

Reviewer 2: Discussed offline: there is currently a problem in invoking the NDK on M1s. The current workaround is listed here: https://reactnative.dev/docs/new-architecture-troubleshooting#android-ndk-and-mac-with-m1-apple-silicon-cpus (I would link against it).

I hope to be able to include this: 
- https://github.com/facebook/react-native/pull/33588

inside a point release so we won't need a workaround anymore

Reviewer 2: Closing as this should be solved with 0.68.1 👍

> In `android/gradle.properties`, enable the new architecture:

Reviewer 2: ```suggestion
In `android/gradle.properties`, enable the New Architecture:
```

```

> newArchEnabled=true

Reviewer 2: We could mention here that you can also enable this property with an environment variable (maybe useful for some users on CI?). It's explained here:
https://reactnative.dev/blog/2022/03/15/an-update-on-the-new-architecture-rollout#the-new-architecture-template

```

> Then build and run the app as usual:

Reviewer 2: Commenting here but is more of a general comment:
I think mention that once the app is running on New Arch, users will see on Metro `fabric=true`. We need give the user a visual feedback on what to expect when enabling the New Architecture.

```shell

> npx react-native run-android

Reviewer 2: On Android the build time between old arch and new arch will have a significant difference. I would probably add a note linking to this page https://reactnative.dev/docs/build-speed on how to improve the build speed if the build is too slow
